test/CometWrapper.t.sol

```diff
@@ -75,7 +80,185 @@ contract CometWrapperTest is BaseTest, CometMath {
         assertEq(cometWrapper.totalAssets(), comet.balanceOf(wrapperAddress));
     }
 
-    function test__nullifyInflationAttacks() public {
+    function test_underlyingBalance() public {
+        assertEq(cometWrapper.underlyingBalance(alice), 0);
+
+        vm.startPrank(alice);
+        comet.allow(wrapperAddress, true);
+        cometWrapper.deposit(5_000e6, alice);
+        vm.stopPrank();
+
+        assertApproxEqAbs(cometWrapper.underlyingBalance(alice), 5_000e6, 1);
+        skip(14 days);
+        assertGe(cometWrapper.underlyingBalance(alice), 5_000e6);
+    }
+
+    function test_previewDeposit() public {
+        assertEq(cometWrapper.balanceOf(alice), 0e6);
+
+        uint256 aliceCometBalance = comet.balanceOf(alice);
+        uint256 alicePreviewedSharesReceived = cometWrapper.previewDeposit(5_000e6);
+        uint256 aliceSharesFromAssets = cometWrapper.convertToShares(5_000e6);
+
+        vm.startPrank(alice);
+        comet.allow(wrapperAddress, true);
+        uint256 aliceActualSharesReceived = cometWrapper.deposit(5_000e6, alice);
+        vm.stopPrank();
+
+        assertApproxEqAbs(comet.balanceOf(alice), aliceCometBalance - 5_000e6, 1);
+        assertEq(cometWrapper.balanceOf(alice), alicePreviewedSharesReceived);
+        assertEq(alicePreviewedSharesReceived, aliceActualSharesReceived);
+        assertEq(alicePreviewedSharesReceived, aliceSharesFromAssets);
+
+        assertEq(cometWrapper.balanceOf(bob), 0e6);
+
+        uint256 bobCometBalance = comet.balanceOf(bob);
+        uint256 bobPreviewedSharesReceived = cometWrapper.previewDeposit(5_000e6);
+        uint256 bobSharesFromAssets = cometWrapper.convertToShares(5_000e6);
+
+        vm.startPrank(bob);
+        comet.allow(wrapperAddress, true);
+        uint256 bobActualSharesReceived = cometWrapper.deposit(5_000e6, bob);
+        vm.stopPrank();
+
+        assertApproxEqAbs(comet.balanceOf(bob), bobCometBalance - 5_000e6, 1);
+        // TODO: investigate rounding
```

Comment:
is this value different than what you expected?

Reply:
I expect there to be some rounding, so this is just a note for me to look into it deeper and make sure the rounding makes sense.